docs/custom-metric-filter.md

```diff
@@ -0,0 +1,195 @@
+  pipelines:
+    # This pipeline exports to Cloud Monitoring backend and enables
+    # customer in-project monitoring. If custom is in need of disabling the
+    # Cloud Monitoring, this section can be removed.
+    metrics/cloudmonitoring:
+      receivers: [opencensus]
+      processors: [batch, filter/cloudmonitoring]
+      exporters: [googlecloud]
+    # This pipeline exports to Prometheus backend, follow
+    # http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#prometheus
+    # to setup
+    metrics/prometheus:
+      receivers: [opencensus]
+      processors: [batch]
+      exporters: [prometheus]
+    ...
+```
+
+## Patch the otel collector deployment with Configmap
+
+### Get the config map
+
+```
+kubectl get cm otel-collector-googlecloud -n config-management-monitoring -o yaml > otel-config.yaml
+```
+
+Or in ACM with version prior to `v1.11.1`, use
+
+```
+kubectl get cm otel-collector-stackdriver -n config-management-monitoring -o yaml > otel-config.yaml
+```
+
+### Modify the otel-config.yaml file
+
+1.  To **add or drop metrics** from the
+    [available metrics](http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#metrics),
+    modify the `include` section under the
+    [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor)
+    configuration
+
+    > Note: If running ACM version prior to 1.11.x, use the
+    [default config](https://github.com/tiffanny29631/kpt-config-sync/blob/main/pkg/metrics/otel.go#L38)
+
+    ```
+    # This is an example adding crd_count metric
+    # Make sure you are modifying the `filter/cloudmonitoring`
+    ...
+    processors:
+      ...
+      filter/cloudmonitoring:
+        metrics:
+          include:
+            match_type: regexp
+            metric_names:
+              - reconciler_errors
+              - pipeline_error_observed
+              - declared_resources
+              - apply_operations_total
+              - resource_fights_total
+              - internal_errors_total
+              - kcc_resource_count
+              - resource_count
+              - ready_resource_count
+              - cluster_scoped_resource_count
+              - resource_ns_count
+              - api_duration_seconds
+              - crd_count         <----- Here
+    ```
+
+1.  To **turn off the custom metrics** in Cloud Monitoring, remove this section
+
+    Note: In ACM version prior to 1.12.0, the exporter name was `stackdriver`
+    instead of `googlecloud`.
+
+    ```
+    metrics/cloudmonitoring:
+      receivers: [opencensus]
+      processors: [batch, filter/cloudmonitoring]
+      exporters: [googlecloud]
+    ```
+
+### Apply the config map
+
+```
+kubectl apply -f otel-config.yaml
+```
+
+### Restart the Otel Collector pod to pick up the new version of Configmap
+
+```
+# This can be done by deleting the Otel Collector pod
+kubectl delete <POD_NAME> -n config-management-monitoring
```

Comment:
you could add labels here so you dont need to know the pod name. Or use kubectl rollout.

Reply:
Changed to locate by deployment